api/src/main/java/org/infinispan/api/common/events/ListenerHandle.java

```diff
@@ -0,0 +1,10 @@
+package org.infinispan.api.common.events;
+
+/**
+ * @since 13.0
+ **/
+public interface ListenerHandle<T> {
+   T get();
+
+   void remove();
```

Comment:
This should probably return a `CompletionStage<Void>` as this will most likely require network calls.

Reply:
This is being discussed below.